ludwig/progress_bar.py

```diff
@@ -0,0 +1,48 @@
+import os
+import uuid
+
+import ray.train
```

Comment:
Ray is technically an optional import. Since this progress bar is on the critical import path, we should move it. There are a few ways to accomplish this. The one that would require the least refactoring would be to do something like this:

```
try:
    import ray.train as rt
except ImportError:
    rt = None

...

class LudwigProgressBar:
    def __init__(...):
        if report_to_ray and rt is None:
            raise ValueError("Set report_to_ray=True but ray is not installed. Run `pip install ray`")
```